src/layouts/main/styles/functions.js

```diff
@@ -0,0 +1,74 @@
+import {css} from 'styled-components';
+import {drawerWidth} from './constants';
+
+/**
+ * To return drawer styles when is open.
+ *
+ * @param {object} theme
+ * @return {css} drawer styles when opened
+ */
+const getOpenDrawerStyles = (theme) => css`
+  width: ${drawerWidth}px;
+  transition: ${theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  })};
+`;
+
+/**
+ * To return drawer styles when is closed.
+ *
+ * @param {object} theme
+ * @return {css} drawer styles when closed
+ */
+const getClosedDrawerStyles = (theme) => css`
+  transition: ${theme.transitions.create('width', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  })};
+  overflow-x: hidden;
+  width: ${theme.spacing(9) + 1}px;
```

Comment:
Curious why these numbers were chosen and why adding 1?

Reply:
tbh, I copied some style examples from the official docs, and adapted accordingly, I removed the + 1 since it actually has no purpose.

Regarding the spacing, the spec is [here](https://material-ui.com/customization/spacing/), but there is no real motivation to select `theme.spacing(9)`, just an opinionated decision from my side.

Let me know if you have a clue on how we can organize these decisions better.